### PR TITLE
Add deterministic autoscaling management

### PR DESCRIPTION
**Pull request by delvh**
Add support for 
- increasing memory to a custom percentage
- clearing the created memory leak
- triggering deterministic SLO violations

Add support but it is not working as expected for
- retrieving memory stats (returns `500 - No handler found`?)

Currently, there is no support for
- setting CPU usage

### CODE REVIEW

File: src/main/java/de/unistuttgart/t2/common/scaling/MemoryInfo.java [hunk 1]
```diff
@@ -0,0 +1,16 @@
+package de.unistuttgart.t2.common.scaling;
+
+public class MemoryInfo {
```
Comment (nk-coding) [inline src/main/java/de/unistuttgart/t2/common/scaling/MemoryInfo.java:3]: missing documentation

**Review by nk-coding — CHANGES_REQUESTED** [review]
Your formatting is inconsistent with the rest of the project. Please change the formatting of the whole project or adapt your changes

**Comment by delvh** [comment]
As we will likely have the formatting discussion more often:
Is there perhaps an official Eclipse formatter I can use for the T2 Projects?
Or should I simply format according to my formatter (which will change **a lot**)?

File: src/main/java/de/unistuttgart/t2/common/scaling/RequestDenier.java [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+ */
+public final class RequestDenier implements HandlerInterceptor {
+
+    private static volatile boolean blockRoutes;
+
+    @Override
+    public boolean preHandle(HttpServletRequest request, HttpServletResponse response,
+        Object handler) throws Exception {
+        if (blockRoutes)
+            response.sendError(HttpStatus.SERVICE_UNAVAILABLE.value());
```
Comment (nk-coding) [inline src/main/java/de/unistuttgart/t2/common/scaling/RequestDenier.java:26]: please always use brackets for if, while, ...

File: src/test/java/de/unistuttgart/t2/common/scaling/MemoryLeakerTest.java [hunk 3]
```diff
@@ -0,0 +1,45 @@
+package de.unistuttgart.t2.common.scaling;
```
[the remark this answers is not in the document]
Comment (delvh) [reply src/test/java/de/unistuttgart/t2/common/scaling/MemoryLeakerTest.java:1]: Yep, forgot the test.

**Review by nk-coding — APPROVED** [review]

**Pull request merged**